in_toto/keylib.go

```diff
@@ -0,0 +1,137 @@
+package in_toto
+
+import (
+  "os"
+  "fmt"
+  "encoding/pem"
+  "encoding/hex"
+  "io/ioutil"
+  "crypto"
+  "crypto/rsa"
+  "crypto/x509"
+  "crypto/sha256"
+  "strings"
+  "reflect"
+)
+
+func (k *Key) LoadPublicKey(path string) error {
+  keyFile, err := os.Open(path)
+  defer keyFile.Close()
+  if err != nil {
+    return err
+  }
+
+  // Read key bytes and decode PEM
+  keyBytes, err := ioutil.ReadAll(keyFile)
+  if err != nil {
+    return err
+  }
+
+  // TODO: There could be more key data in _, which we silently ignore here.
+  // Should we handle it / fail / say something about it?
+  data, _ := pem.Decode([]byte(keyBytes))
+  if data == nil {
+    return fmt.Errorf("No valid public rsa key found at '%s'", path)
+  }
+
+  // Try if this is a indeed a public key, just to see if the key is valid
+  _, err = x509.ParsePKIXPublicKey(data.Bytes)
```

Comment:
It does not, `x509.ParsePKIXPublicKey` returns an `interface` that may be a struct of type `rsa.PublicKey`, `dsa.PublicKey`, or `ecdsa.PublicKey`, but we need to store the key in the original pem format, extracted by the next ~15 lines.

I only call `pem.Decode` and `x509.ParsePKIXPublicKey` to fail early if the data can't be parsed. Does that make sense?

I could reconstruct the original pem, by using the data returned by one of those two functions, but it would not reduce code.